src/goto-symex/goto_symex.h

```diff
@@ -91,18 +111,28 @@ class goto_symext
   {
   }
 
+  /// A virtual destructor allowing derived classes to be cleaned up correctly
   virtual ~goto_symext() = default;
 
+  /// The type of delegate functions that retrieve a goto_functiont for a
+  /// particular function identifier
+  /// \remarks
+  /// This allows SymEx to be divorced from the particular type of goto_modelt
+  /// that provides the function bodies
   typedef
     std::function<const goto_functionst::goto_functiont &(const irep_idt &)>
     get_goto_functiont;
 
   /// \brief symex entire program starting from entry point
-  ///
-  /// The state that goto_symext maintains has a large memory footprint.
-  /// This method deallocates the state as soon as symbolic execution
-  /// has completed, so use it if you don't care about having the state
-  /// around afterwards.
+  /// \remarks
+  /// The state that goto_symext maintains uses a lot of memory.
+  /// This method therefore deallocates the state as soon as symbolic execution
+  /// has completed. This function is useful to callers that don't care about
+  /// having the state around afterwards.
+  /// \param get_goto_function The delegate to retrieve function bodies (see
+  /// \ref get_goto_functiont)
+  /// \param new_symbol_table A symbol table to store the symbols added by
+  /// SymEx
```

Comment:
This camel-casing makes it sound like there were one particular thing of that name. Maybe just say "store the symbols adding during symbolic execution" (applies here and elsewhere)